### PR TITLE
feat(order): add state root slice

### PR DESCRIPTION
**Pull request by griest024**
BREAKING CHANGES: selector types have changed

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

Fixes: #1595

## Does this PR introduce a breaking change?
```
[x] Yes
[ ] No
```

selector types have changed

## Other information

### CODE REVIEW

**Comment by damienwebdev** [comment]
@griest024 can you split the two concerns of this PR into separate PRs?

**Comment by griest024** [comment]
> @griest024 can you split the two concerns of this PR into separate PRs?

#1623

**Review by lderrickable — CHANGES_REQUESTED** [review]

File: libs/order/state/src/facades/order/order.facade.spec.ts [hunk 1]
```diff
@@ -34,7 +34,7 @@ import {
 import { DaffOrderFacade } from './order.facade';
 
 describe('DaffOrderFacade', () => {
-  let store: Store<{ [DAFF_ORDER_STORE_FEATURE_KEY]: Partial<DaffOrderReducersState> }>;
+  let store: Store<{ [DAFF_ORDER_STORE_FEATURE_KEY]: Partial<DaffOrderStateRootSlice> }>;
```
[the remark this answers is not in the document]
Comment (griest024) [reply libs/order/state/src/facades/order/order.facade.spec.ts]: yes and actually I think most of the facade specs have this issue

[the remark this answers is not in the document]
Comment (griest024) [reply libs/order/state/src/facades/order/order.facade.spec.ts]: fixed

**Review by damienwebdev — CHANGES_REQUESTED** [review]

File: libs/order/state/src/facades/order/order.facade.spec.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (griest024) [reply libs/order/state/src/facades/order/order.facade.spec.ts]: yeah I actually removed it already :joy:

**Review by lderrickable — APPROVED** [review]

File: libs/order/state/src/order-state.module.ts [hunk 2]
```diff
@@ -10,6 +12,7 @@ import {
 
 @NgModule({
   imports: [
+    DaffCartStateModule,
```
Comment (damienwebdev) [inline libs/order/state/src/order-state.module.ts:15]: Is this intentional?

**Review by damienwebdev — CHANGES_REQUESTED** [review]

File: libs/order/state/src/order-state.module.ts (hunk 2 above)
Comment (griest024) [reply libs/order/state/src/order-state.module.ts:15]: yes, order state has a direct dep on cart state

**Pull request merged**